From david-b@pacbell.net  Wed Apr 22 09:46:31 2009
From: Robert Jarzmik <robert.jarzmik@free.fr>
From: David Brownell <david-b@pacbell.net>
Date: Tue, 21 Apr 2009 20:34:24 -0700
Subject: USB: pxa27x_udc: introduce pxa27x_clear_otgph()
To: Robert Jarzmik <robert.jarzmik@free.fr>
Cc: Greg KH <greg@kroah.com>
Message-ID: <200904212034.24768.david-b@pacbell.net>


Follow pxa27x change in OTGPH handling, and use the newly
defined pxa27x_clear_otgph().

Signed-off-by: Robert Jarzmik <robert.jarzmik@free.fr>
Acked-by: Eric Miao <eric.miao@marvell.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>


---
 drivers/usb/gadget/pxa27x_udc.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

--- a/drivers/usb/gadget/pxa27x_udc.c
+++ b/drivers/usb/gadget/pxa27x_udc.c
@@ -38,7 +38,6 @@
 #include <linux/usb.h>
 #include <linux/usb/ch9.h>
 #include <linux/usb/gadget.h>
-#include <mach/pxa2xx-regs.h> /* FIXME: for PSSR */
 #include <mach/udc.h>
 
 #include "pxa27x_udc.h"
@@ -2479,6 +2478,12 @@ static void pxa_udc_shutdown(struct plat
 		udc_disable(udc);
 }
 
+#ifdef CONFIG_CPU_PXA27x
+extern void pxa27x_clear_otgph(void);
+#else
+#define pxa27x_clear_otgph()   do {} while (0)
+#endif
+
 #ifdef CONFIG_PM
 /**
  * pxa_udc_suspend - Suspend udc device
@@ -2546,8 +2551,7 @@ static int pxa_udc_resume(struct platfor
 	 * Software must configure the USB OTG pad, UDC, and UHC
 	 * to the state they were in before entering sleep mode.
 	 */
-	if (cpu_is_pxa27x())
-		PSSR |= PSSR_OTGPH;
+	pxa27x_clear_otgph();
 
 	return 0;
 }
